### DISCUSSION TITLE
TLS key negotiation failed to occur within 60 seconds

### DISCUSSION THREAD

**Discussion by gilbrotheraway** [pivpn/pivpn#1531] [Support]
## In raising this issue, I confirm the following:

`{please fill the checkboxes, e.g: [X]}`

- X] I have read the [documentation](https://docs.pivpn.io)
- [] Is it a feature request? please consider opening a [Discussion] (https://github.com/pivpn/pivpn/discussions/new)
- [] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.  
- [] The issue I am reporting *is* directly related to the pivpn installer script.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

### Has your install failed? 
```
no
```

### Describe the issue
it gets stuck on this 👍 

MANAGEMENT: >STATE:1651040955,WAIT,,,,,,

**Expected behavior**
use the vpn on windows

### Can you replicate the issue? Describe the steps below

1. Go to openvpn or wireguard
2. Click on 'add tunnel or config file'

### Have you searched for similar issues and solutions?
    YES!!

**Additional context**
wireguard was working few weeks back now i did the same exact install and nothing works

### Have you taken any steps towards solving your issue?
```
 reinstalled multiple times, tried the dev branch described on first page, changed firewall rules, etc
```

#### What type of hardware are you running PiVPN at? 

```
Raspberrypi 3b+ running raspbian buster custom OS

#### Output of `uname -a`
```
Linux raspberrypi 5.10.103-v7+ #1529 SMP Tue Mar 8 12:21:37 GMT 2022 armv7l GNU/Linux

```

#### Output of `cat /etc/os-release`
```

PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

```
### Output of -Debug

all green no errors

######

it gets stuck on 
 UDP link remote: [AF_INET]********:1194
STATE:1651041365,WAIT,,,,,,
TLS Error: TLS key negotiation failed to occur within 60 seconds (check your network connectivity)
TLS handshake failed


im really tires been many hours trying to setup this

**Comment by gilbrotheraway** [comment]
it works on android(3g) but tcpdump confirmed it doesn´t on windows

**Comment by gilbrotheraway** [comment]
ok so when i set remote directive on client .ovpn file to local ip it connects, but then i have no internet, should i go on with this approach? is it safe?

**Comment by orazioedoardo** [comment]
> ok so when i set remote directive on client .ovpn file to local ip it connects

You are connecting to the Pi next to you, it's quite useless. You are supposed to connect to the PiVPN remotely. Can you try to connect the Windows PC via Android hotspot and then to PiVPN? Also please post the debug log `pivpn -d`.

**Comment by coolapso** [comment]
Moving to discussions .. seems this topic has died.
